src/meta/meta_turtle.c

```diff
@@ -81,10 +83,46 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
             WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+
         WT_ERR(__wt_metadata_update(session, key->data, value->data));
     }
 
     F_SET(S2C(session), WT_CONN_WAS_BACKUP);
+    WT_ERR(__wt_fclose(session, &fs));
+    WT_ERR(__wt_fopen(session, WT_METADATA_BACKUP, 0, WT_STREAM_READ, &fs));
+    if (F_ISSET(S2C(session), WT_CONN_BACKUP_PARTIAL)) {
+        for (;;) {
+            WT_ERR(__wt_getline(session, fs, key));
+            if (key->size == 0)
+                break;
+            WT_ERR(__wt_getline(session, fs, value));
+            if (value->size == 0)
+                WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+
+            filename = (char *)key->data;
+            if (WT_PREFIX_SKIP(filename, "file:")) {
+                WT_ERR(__wt_fs_exist(session, filename, &exist));
+                if (!exist) {
+                    WT_ERR(__wt_metadata_cursor(session, &cursor));
+                    while (cursor->next(cursor) == 0) {
+                        cursor->get_key(cursor, &key_metadata);
+                        prefix_skipped_metadata = key_metadata;
+                        if (WT_PREFIX_SKIP(prefix_skipped_metadata, "file:") ||
```

Comment:
We definitely want to use schema operations here because that will do all the table: -> colgroup: -> file: translations for you.